xbmc/utils/test/TestCPUInfo.cpp

```diff
@@ -53,6 +53,17 @@ TEST_F(TestCPUInfo, GetTemperature)
   EXPECT_TRUE(t.IsValid());
 }
 
+#if defined(TARGET_DARWIN_OSX)
```

Comment:
This needs to be disabled if we are building for x86_64 but the tests are running on arm (otherwise we try to read non-existing smc sensors). Any ideas how I can ifdef this?

Reply:
just add an and condition to cpu check it

https://github.com/xbmc/xbmc/blob/a4f66924c3c5ddf6fcd0e0038020ff94cffbd213/xbmc/addons/addoninfo/AddonInfoBuilder.cpp#L859

Reply:
Won't this disable the test if running on arm but building for arm as well?